systemservices/marketplace/src/tasks/preparePublishServiceVersion.ts

```diff
@@ -1,19 +1,38 @@
 import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
 import { Marketplace } from "../contracts/Marketplace"
-import { stringToHex, CreateTransaction, hashToHex } from "../contracts/utils";
+import { stringToHex, CreateTransaction } from "../contracts/utils";
 import { Manifest } from "../types/manifest";
+import { getService, isServiceExist } from "../contracts/service";
 
 export default (
   marketplace: Marketplace,
   createTransaction: CreateTransaction
 ) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
   try {
+    // check inputs
+    const sid = inputs.service.definition.sid
+    if (!(1 <= sid.length && sid.length <= 63)) throw new Error('sid must have a length between 1 and 63')
```

Comment:
the `63` should be in a constant somewhere to know what is this 63
the condition is a bit complicated. Either invert it `sid.length < 1 || sid.length > 63` or even 2 if with their own message, one for min length, one for max length

Reply:
I simplify the test but leave the 63 hardcoded because the go implementation is also hardcoded.
Anyway I think this limitation will be removed in the future, so I don't want to spend too much time on this test.

Reply:
> because the go implementation is also hardcoded

haha, this is the worst argument ever but anyway it's not blocking at all